### PR TITLE
Add support for git submodules. Addresses issue #54.

### PR DESCRIPTION
**Pull request by cojomojo**

### CODE REVIEW

File: lib/command-requirements.js [hunk 1]
```diff
@@ -5,12 +5,20 @@ module.exports = {
     var stat;
     try {
       stat = fs.statSync(req.path + '/.git/');
-      if (stat.isDirectory()) {
-        return true;
+      if (stat.isDirectory()) return true;
+      else {
+        // check if the directory is a submodule
+        // addresses https://github.com/mixu/gr/issues/54
+        var path = req.path.split(/(\\|\/)/).slice(0, -2).join('/');
```
Comment (mixu) [inline lib/command-requirements.js]: `path.dirname(path.dirname(req.path))`?

File: lib/command-requirements.js [hunk 2]
```diff
@@ -5,12 +5,20 @@ module.exports = {
...
+        stat = fs.statSync(path + '/.git/');
+        if (stat.isDirectory()) {
+          stat = fs.statSync(path + '/.gitmodules')
+          if (stat.isFile()) return true;
```
Comment (mixu) [inline lib/command-requirements.js]: nit: needs curly braces

**Comment by cojomojo** [comment]
@mixu I updated the code according to your suggestions, and moved it into the catch block since statSync *should* throw an error when trying to access a non-existent directory. I say should because of https://github.com/nodejs/node/issues/9106.

**Comment by mixu** [comment]
:+1: thanks for making these changes and for the PR! Merged!

**Comment by mixu** [comment]
published as `v0.5.2`

**Comment by cojomojo** [comment]
@mixu you're welcome, thanks for the great tool!

**Pull request merged**